src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js

```diff
@@ -80,34 +81,13 @@ const DomainList = (props) => {
   };
 
   const renderProfilePicture = () => {
-    if (item.karmaScore) {
-      return (
-        <CircleGradient
-          testId="images"
-          fill={item.karmaScore}
-          size={normalize(51)}
-          width={normalize(3)}>
-          <Image
-            testId="images"
-            source={{
-              uri: item.image
-            }}
-            style={styles.profilepicture}
-            width={48}
-            height={48}
-          />
-        </CircleGradient>
-      );
-    }
     return (
-      <Image
-        testId="images"
-        source={{
-          uri: item.image
-        }}
-        style={styles.profilepicture}
-        width={48}
-        height={48}
+      <ProfilePicture
+        profilePicPath={item.image}
+        karmaScore={item.karmaScore}
+        size={51}
+        width={3}
+        withKarma={withKarma}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `renderProfilePicture` function is more concise and maintainable. It uses the `ProfilePicture` component and passes necessary props including `withKarma` for conditional rendering of the Karma score. 

```diff
-    if (item.karmaScore) {
-      return (
-        <CircleGradient
-          testId="images"
-          fill={item.karmaScore}
-          size={normalize(51)}
-          width={normalize(3)}>
-          <Image
-            testId="images"
-            source={{
-              uri: item.image
-            }}
-            style={styles.profilepicture}
-            width={48}
-            height={48}
-          />
-        </CircleGradient>
-      );
-    }
-     return (
-      <Image
-        testId="images"
-        source={{
-          uri: item.image
-        }}
-        style={styles.profilepicture}
-        width={48}
-        height={48}
-       />
-     );
+    return (
+      <ProfilePicture
+        profilePicPath={item.image}
+        karmaScore={item.karmaScore}
+        size={51}
+        width={3}
+        withKarma={withKarma}
+      />
+    );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->